compiler/rustc_ast_passes/src/ast_validation.rs

```diff
@@ -1002,6 +995,7 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
                     let forbidden_let_reason = Some(ForbiddenLetReason::ForbiddenWithOr(*span));
                     this.with_let_management(forbidden_let_reason, |this, _| this.visit_expr(lhs));
                     this.with_let_management(forbidden_let_reason, |this, _| this.visit_expr(rhs));
+                    return;
```

Comment:
Am I misreading the surrounding code, or does this change not have any semantic effect?

And in any case, why is this change in this PR?

Reply:
This line was an oversight that is now removed